arcaflow_plugin_coremark_pro/coremark_pro_plugin.py

```diff
@@ -116,16 +136,25 @@ def certify_all(
     }
 
     for line in ca_return[1].splitlines():
+        line_list = line.split()
         try:
-            line_name = line.split()[0]
+            line_name = line_list[0]
         except IndexError:
             pass
-        if line_name in ca_results and len(line.split()) > 1:
+        if line_name in ca_results and len(line_list) > 1:
             ca_results[line_name] = {
-                "MultiCore": line.split()[1],
-                "SingleCore": line.split()[2],
-                "Scaling": line.split()[3],
+                "MultiCore": line_list[1],
+                "SingleCore": line_list[2],
+                "Scaling": line_list[3],
             }
+            if line_name != "CoreMark-PRO":
+                with open(file=run_log_path, encoding="utf-8") as file:
+                    for line in file:
+                        line_list = line.split()
+                        if "median single" in line and line_list[2] == line_name:
+                            ca_results[line_name]["Iterations"] = int(
+                                line_list[7]
+                            )
```

Comment:
Should this be dependent on the `len(line_list) > 1` test up at line 144? The block doesn't reference `line_list` (at least, until it overwrites that with the log file contents -- which isn't ideal). Although obviously that may not matter if the `CoreMark-PRO` line always has parameters.

Reply:
I think that length check was solving a problem resulting from "I'm too tired to be working on this, but I'm going to keep going anyway."

Reply:
> I think that length check was solving a problem resulting from "I'm too tired to be working on this, but I'm going to keep going anyway."

A tried and true implementation technique, for sure -- I find myself using close variants of that often. 😀